packages/pwa-kit-cli/src/configs/webpack/config.js

```diff
@@ -102,7 +102,9 @@ const baseConfig = (target) => {
                     minimize: mode === production
                 },
                 // Always use source map, makes debugging the server much easier.
-                devtool: 'source-map',
+                // Use eval-source-map for server debugging in development mode
+                devtool:
+                    mode === development && target === 'node' ? 'eval-source-map' : 'source-map',
```

Comment:
The base webpack config object is applied to all of the 5 configs:
- client bundles (like main.js and chunks page-home.js)
- ssr.js
- server-renderer.js
- client optionals (like worker.js, polyfills)
- request-processor.js

I'm wondering if the devtool settings is applicable to all of the configs?

Reply:
For client side bundles, we definitely don't want to have inline source maps, because that increases bundle size.

Reply:
For server side, there are two modes we care about, `development` and `production`.

For `development`, we'd like to have inline source map due to Chrome devtools not able to fetch external sourcemaps.

What about `production` mode, do we want to have sourcemap in production for ssr.js and server-renderer.js? I's say we probably don't need sourcemap in this case.

Reply:
seems like the `devtool` option varies based on the `target`. Instead of adding a complex ternary logic in the base webpack config, have you thought about adding `devtool` to the individual webpack configs?

Reply:
Another thought that is probably out of scope for this PR... 

Should we not ship sourcemaps for production? or make it optional? Today, the sourcemaps are shipped to production, which means everybody can view the source code and I'd imagine that some people want to disable sourcemap for security reasons.

Reply:
devtool is in the baseConfig which is called from each config where it will pass in the target, so yes it will be applicable for all configs. For example
```
const renderer =
    fs.existsSync(path.resolve(projectDir, 'node_modules', 'pwa-kit-react-sdk')) &&
    baseConfig('node')
```

We certainly don't need a source map for serverside on prod, but I think we do still want to have source map for client. So we have
- mode: dev - client side ( target: web) => source-map
- mode: dev - server-sider (target: node) => eval-source-map
- mode: prod - client and server (I dont think we can debug serverside on prod) => source-map
In other words, if the config is for server-sider, and in development, use `eval-source-map`. The rest should keep using `source-map` as it was. 

I don't think it is complex that requires us to break the devtool into each config.

Reply:
@kevinxh That is a good question, I originally thought it is not necessary to have source-map on prod too, but I believe there was a historical context that we don't know about. @bendvc Would you mind filling us in on the reason we need source-map for prod env?

Reply:
@alexvuong For prod + server combination, this ternary will evaluate to 'source-map', which will produce ssr.js.map and server-renderer.js.map, which is useless and make the build zip file larger.

Reply:
we don't need sourcemap for request-processor.js for production either.

Reply:
Good point. I've made a change to add source map to two configs: client and renderer config. I don't think we need source-map for `client-optional`. 🤔 . Besides, ssr.js is not compiled in prod, so it does not need source-map so to speak.